### PR TITLE
CXX-636: make return of write concern tag an optional

### PR DESCRIPTION
**Pull request by Machyne**

### CODE REVIEW

File: src/mongocxx/write_concern.cpp [hunk 1]
```diff
@@ -17,13 +17,15 @@
...
+#include <bsoncxx/stdx/optional.hpp>
+#include <mongocxx/stdx.hpp>
 #include <mongocxx/private/libmongoc.hpp>
 
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 
 write_concern::write_concern()
-    : _impl{bsoncxx::stdx::make_unique<impl>(mongoc_write_concern_new())} {
+    : _impl{stdx::make_unique<impl>(mongoc_write_concern_new())} {
```
Comment (amidvidy) [inline src/mongocxx/write_concern.cpp]: libmongoc::write_concern_new()

File: src/mongocxx/write_concern.cpp [hunk 2]
```diff
@@ -86,10 +88,13 @@ std::int32_t write_concern::nodes() const {
     return libmongoc::write_concern_get_w(_impl->write_concern_t);
 }
 
-// TODO: should this be an optional... probably
-std::string write_concern::tag() const {
+stdx::optional<std::string> write_concern::tag() const {
+    stdx::optional<std::string> to_return;
     const char* tag_str = libmongoc::write_concern_get_wtag(_impl->write_concern_t);
-    return tag_str ? tag_str : std::string();
+    if (tag_str) {
```
Comment (amidvidy) [inline src/mongocxx/write_concern.cpp]: return tag_str ? : stdx::make_optional<std::string>(tag_str) : stdx::nullopt_t

**Comment by amidvidy** [comment]
LGTM :+1:

**Comment by Machyne** [comment]
(merged)

**Pull request closed without merging**